### PR TITLE
[Bug](materialized-view) add limit for drop column on mv

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes
add limit for drop column on mv

```java
2023-09-15 20:08:01,085 WARN (leaderCheckpointer|102) [MaterializedIndexMeta.parseStmt():310] CreateMaterializedViewStmt analyze failed, mv=create materialized view mv1 as select user_id, date, city, age, sum(cost) from schema_change_agg_mv_regression_test group by u
ser_id, date, city, age;, reason=
org.apache.doris.common.AnalysisException: errCode = 2, detailMessage = Unknown column 'cost' in 'table list'
        at org.apache.doris.common.ErrorReport.reportAnalysisException(ErrorReport.java:53) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.common.ErrorReport.reportAnalysisException(ErrorReport.java:48) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.analysis.Analyzer.registerColumnRef(Analyzer.java:980) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.analysis.SlotRef.analyzeImpl(SlotRef.java:204) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.analysis.Expr.analyze(Expr.java:443) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.analysis.Expr.analyze(Expr.java:435) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.analysis.SelectStmt.analyze(SelectStmt.java:537) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.analysis.CreateMaterializedViewStmt.analyze(CreateMaterializedViewStmt.java:168) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.catalog.MaterializedIndexMeta.parseStmt(MaterializedIndexMeta.java:308) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.catalog.OlapTable.analyze(OlapTable.java:2328) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.catalog.Database.analyze(Database.java:616) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.datasource.InternalCatalog.loadDb(InternalCatalog.java:3136) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.catalog.Env.loadDb(Env.java:1848) ~[doris-fe.jar:1.2-SNAPSHOT]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_131]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_131]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_131]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_131]
        at org.apache.doris.persist.meta.MetaReader.read(MetaReader.java:116) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.catalog.Env.loadImage(Env.java:1779) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.master.Checkpoint.doCheckpoint(Checkpoint.java:151) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.master.Checkpoint.runAfterCatalogReady(Checkpoint.java:79) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.common.util.MasterDaemon.runOneCycle(MasterDaemon.java:58) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.common.util.Daemon.run(Daemon.java:116) ~[doris-fe.jar:1.2-SNAPSHOT]
2023-09-15 20:08:01,087 WARN (leaderCheckpointer|102) [MaterializedIndexMeta.parseStmt():310] CreateMaterializedViewStmt analyze failed, mv=create materialized view mv2 as select user_id, date, city, age, cost from schema_change_uniq_mv_regression_test;, reason=
org.apache.doris.common.AnalysisException: errCode = 2, detailMessage = Unknown column 'cost' in 'table list'
```
## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeHandler.java [hunk 1]
```diff
@@ -410,12 +411,20 @@ private boolean processDropColumn(DropColumnClause alterClause, OlapTable olapTa
...
-                    if (column.getName().equalsIgnoreCase(dropColName)) {
-                        if (column.isKey()) {
-                            lightSchemaChange = false;
+                    boolean needDrop = column.getName().equalsIgnoreCase(dropColName);
+                    if (!needDrop && column.getDefineExpr() != null) {
+                        List<SlotRef> slots = new ArrayList<>();
+                        column.getDefineExpr().collect(SlotRef.class, slots);
+                        for (SlotRef slot : slots) {
+                            if (slot.getColumnName().equalsIgnoreCase(dropColName)) {
+                                needDrop = true;
```
Comment (HappenLee) [inline fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeHandler.java:420]: needDropMVFirst ? better name?

File: fe/fe-core/src/test/java/org/apache/doris/alter/SchemaChangeHandlerTest.java [hunk 2]
```diff
@@ -197,16 +197,20 @@ public void testAggAddOrDropColumn() throws Exception {
         //process agg drop value column with rollup schema change
         String dropRollUpValColStmtStr = "alter table test.sc_agg drop column max_dwell_time";
         AlterTableStmt dropRollUpValColStmt = (AlterTableStmt) parseAndAnalyzeStmt(dropRollUpValColStmtStr);
-        Env.getCurrentEnv().getAlterInstance().processAlterTable(dropRollUpValColStmt);
-        jobSize++;
```
Comment (HappenLee) [inline fe/fe-core/src/test/java/org/apache/doris/alter/SchemaChangeHandlerTest.java:201]: remove the code is ok ?

**Comment by BiteTheDDDDt** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by zclllyybb — APPROVED** [review]

**Pull request merged**